### DISCUSSION TITLE
Is it possible to auto-rollback after a failed sync?

### DISCUSSION THREAD

**Discussion by hammadzz** [argoproj/argo-cd#6732] [Q&A]
I understand for deployments the best option to auto-rollback is to use [argo-rollouts](https://argoproj.github.io/argo-rollouts/). 

But my question is broader, to include k8s manifest updates such as a change in configmap or k8s secret, etc. As I understand ArgoCD sync will simply do `kubectl apply` to apply the changes in the repo. 

## What is the behaviour when this sync action fails?
1. Does it auto rollback after a retry? 
1. If not does that mean my app stays in a failed state until someone either does an explicit rollback or reverts the changes in the repo and runs sync?
1. Likewise if it was a helm chart instead of a k8s manifest, what would the on failure behaviour be?

**Comment by jgwest** [comment]
1 - No, it's a manual operation [at this time](https://github.com/argoproj/argo-cd/issues/6147).
2 - This would depend on why it failed, some errors will resolve automatically (for example, resource contraints on the k8s cluster that are later resolved). But generally, yes.
3 - The behaviour would be the same: note that [Helm rollback hooks are not supported](https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#helm-hooks), if this is the context of your question.
